services/venice-server/src/main/java/com/linkedin/venice/listener/ReadQuotaEnforcementHandler.java

```diff
@@ -532,6 +518,46 @@ public void handleStoreChanged(Store store) {
     storeQuotaChangeMap.put(store.getName(), store.getReadQuotaInCU());
   }
 
+  /**
+   * If partition assignment is not available for any reason(s) instead of failing quota initialization we will go ahead
+   * and update the quota based on local storage state. Leveraging {@link StorageEngineRepository} partition assignment
+   * information and assuming this replica will be responsible for all traffic for assigned partition(s).
+   */
+  private void handleQuotaUpdateWithoutPartitionAssignment(
+      Store store,
+      int versionNumber,
+      String topic,
+      String reason) {
+    Version version = store.getVersion(versionNumber);
+    if (version != null && version.getStatus().equals(VersionStatus.ONLINE)) {
+      if (quotaInitializationFallbackEnabled) {
+        LOGGER.warn(
+            "Unable to get partition assignment due to: {} for resource: {}, Will try to use fallback strategy to initialize the rate limiter",
+            reason,
+            topic);
+        long quotaInRcu = store.getReadQuotaInCU();
+        double thisNodeQuotaResponsibility = 0;
+        StorageEngine storageEngine = storageEngineRepository.getLocalStorageEngine(topic);
+        if (storageEngine != null) {
+          double assignedPartitionCount = storageEngine.getPartitionIds().size();
+          thisNodeQuotaResponsibility = assignedPartitionCount / (double) version.getPartitionCount();
```

Comment:
Nit, just a heads up, I don't think `Version.getPartitionCount` can be 0 in real, but I do see other place that we have code to handle `0` partition [count](https://github.com/linkedin/venice/blob/2b99017715072ccc9bc397be7533dcd89f648e64/services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java#L5228). Maybe we can still assign `thisNodeQuotaResponsibility` to 0, just try to avoid the divide by 0 exception?